### DISCUSSION TITLE
Does C#/CLR runtime has  something similar to Java 14 "Helpful NullPointerException". https://openjdk.java.net/jeps/358

### DISCUSSION THREAD

**Discussion by madhub** [dotnet/runtime#49960] [General]
Does C#/CLR runtime has  something similar to Java 14 "[Helpful NullPointerException](https://openjdk.java.net/jeps/358)". https://openjdk.java.net/jeps/358
Very useful feature to find exact object throwing Null reference

**Comment by yaakov-h** [comment]
The debugger experience [does](https://github.com/dotnet/runtime/issues/3858#issuecomment-476376311) but the language itself and regular runtime experience does not.

**Reply by madhub** [reply]
Not sure why this feature is not a priority , it greatly helps production code debugging.
Is their any feature request for this ? , we can all up vote to increase the priority

**Reply by huoyaoyuan** [reply]
The full experience of this feature requires debug information, because local variable and chained expression can be transformed in release and optimization.

But at least, the runtime can give information about which member of which type the code is trying to access.
